beacon/validator/src/main/java/tech/pegasys/teku/validator/coordinator/performance/DefaultPerformanceTracker.java

```diff
@@ -300,16 +301,23 @@ private AttestationPerformance calculateAttestationPerformance(
 
     // Pre-process attestations included on chain to group them by
     // data hash to inclusion slot to aggregation bitlist
-    final Map<Bytes32, NavigableMap<UInt64, SszBitlist>> slotAndBitlistsByAttestationDataHash =
-        new HashMap<>();
+    final Map<Bytes32, NavigableMap<UInt64, AttestationBitsAggregator>>
+        slotAndBitlistsByAttestationDataHash = new HashMap<>();
     for (Map.Entry<UInt64, List<Attestation>> entry : attestationsIncludedOnChain.entrySet()) {
+      final Optional<Int2IntMap> committeesSize =
+          Optional.of(spec.getBeaconCommitteesSize(state, entry.getKey()));
       for (Attestation attestation : entry.getValue()) {
         final Bytes32 attestationDataHash = attestation.getData().hashTreeRoot();
-        final NavigableMap<UInt64, SszBitlist> slotToBitlists =
```

Comment:
we could potentially use JMH on this...

Reply:
Doing that, will share the results

Reply:
Done: https://github.com/Consensys/teku/pull/8851#issuecomment-2506724226